### PR TITLE
Ria 5343 upload bail summary

### PR DESCRIPTION
**Pull request by Marcellofabbri**
### JIRA link (if applicable) ###
[RIA-5343](https://tools.hmcts.net/jira/browse/RIA-5343)

### Change description ###
Added business logic to handle new event Upload Bail Summary (a presubmit handler and a postsubmit handler for the new event). 
Updated BailCaseDefinition, DocumentTag, Event and State to reflect the new addition.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java [hunk 1]
```diff
@@ -253,7 +253,10 @@ public enum BailCaseFieldDefinition {
         "detentionFacility", new TypeReference<String>(){}),
     APPLICATION_SUBMISSION_DOCUMENT(
         "applicationSubmissionDocument", new TypeReference<List<IdValue<DocumentWithMetadata>>>(){}),
-    ;
+    BAIL_SUMMARY(
+        "uploadBailSummaryDocs", new TypeReference<List<IdValue<DocumentWithDescription>>>(){}),
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:257]: Please follow the convention camelCase -> UPPERCASE_UNDERSCORE_SEPARATED.

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java [hunk 2]
```diff
@@ -253,7 +253,10 @@ public enum BailCaseFieldDefinition {
...
+    BAIL_SUMMARY_WITH_METADATA(
+        "uploadBailSummaryMetadata", new TypeReference<List<IdValue<DocumentWithMetadata>>>(){});
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:259]: Same as above.

**Review by neha-aggarw — CHANGES_REQUESTED** [review]
Please check naming convention for FieldDefinitions.

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java (hunk 1 above)
Comment (Marcellofabbri) [reply src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:257]: Isn't it already following the convention?
The enum constants are ```BAIL_SUMMARY``` and ```BAIL_SUMMARY_WITH_METADATA``` and their values are ```"uploadBailSummaryDocs"``` and ```"uploadBailSummaryMetadata"``` like all other enum constants

Comment (neha-aggarw) [reply src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:257]: uploadBailSummaryDocs -> UPLOAD_BAIL_SUMMARY_DOCS and uploadBailSummaryMetadata -> UPLOAD_BAIL_SUMMARY_METADATA

**Review by neha-aggarw — APPROVED** [review]
LGTM

**Review by ThomasKKC — APPROVED** [review]
LGTM

**Pull request merged**